internal/cloudsql/instance.go

```diff
@@ -133,37 +128,16 @@ func NewInstance(instance string, client *sqladmin.Service, key *rsa.PrivateKey)
 	return i, nil
 }
 
-// Connect returns a secure, authorized net.Conn to a Cloud SQL instance.
-func (i *Instance) Connect(ctx context.Context) (net.Conn, error) {
+// ConnectInfo returns a map of IP types and a TLS config that can be used to connect to a Cloud SQL instance.
+func (i *Instance) ConnectInfo(ctx context.Context) (map[string]string, *tls.Config, error) {
 	i.resultGuard.RLock()
 	res := i.cur
 	i.resultGuard.RUnlock()
 	err := res.Wait(ctx)
```

Comment:
Somewhat tangential, but why is `Wait` a separate, exported method?

Reply:
This is a bit of a kvg-idiom, but I tend to used exported functions on unexported structs to signal that I shouldn't touch the struct from outside of the structs Exported methods.